packages/react/src/components/PhoneNumberInput/PhoneNumberInput.tsx

```diff
@@ -0,0 +1,121 @@
+import {countries, Country} from './constants';
+import {WithWrapperProps} from '../../models';
+import {composeComponentDisplayName} from '../../utils';
+import Box, {BoxProps} from '../Box';
+import Image from '../Image';
+import InputLabel from '../InputLabel';
+import ListItemIcon from '../ListItemIcon';
+import MenuItem from '../MenuItem';
+import './phone-number-input.scss';
+import OutlinedInput, {OutlinedInputProps as MuiOutlinedInputProps} from '../OutlinedInput';
+
+export interface PhoneNumberInputProps extends BoxProps {
+  OutlinedInputProps?: Omit<MuiOutlinedInputProps, 'id' | 'label' | 'placeholder' | 'type'>;
+  SelectProps?: Omit<MuiSelectProps, 'labelId' | 'id' | 'value' | 'onChange' | 'placeholder'>;
+  defaultCountryCode?: string;
+  onChange?: (value: string) => void;
+  placeholder?: string;
+}
+
+const COMPONENT_NAME: string = 'PhoneNumberInput';
+
+const PhoneNumberInput: ForwardRefExoticComponent<PhoneNumberInputProps> & WithWrapperProps = forwardRef(
+  (props: PhoneNumberInputProps, ref: MutableRefObject<HTMLDivElement>): ReactElement => {
+    const {
+      className,
+      defaultCountryCode,
+      label,
+      InputLabelProps,
+      OutlinedInputProps,
+      onChange,
+      placeholder,
+      SelectProps,
+      ...rest
+    } = props;
+
+    const classes: string = clsx('oxygen-phone-number-input', className);
+
+    const [countryCode, setCountryCode] = useState<string>(defaultCountryCode ?? countries[0].dial_code);
+    const [phoneNumber, setPhoneNumber] = useState('');
+
+    const handleCountryCodeChange = (event: SelectChangeEvent): void => {
+      setCountryCode(event.target.value);
+      onChange(`${event.target.value}${phoneNumber}`);
+    };
+
+    const handlePhoneNumberChange = (event: ChangeEvent<HTMLInputElement>): void => {
+      setPhoneNumber(event.target.value);
+      onChange(`${countryCode}${event.target.value}`);
+    };
+
+    return (
+      <Box className={classes} {...rest} ref={ref}>
+        <InputLabel htmlFor="phone-number-input" id="phone-number-label">
+          {label}
+        </InputLabel>
+        <Box className="oxygen-select-input">
+          <Select
+            labelId="phone-number-label"
+            id="phone-number-select"
+            value={countryCode}
+            onChange={handleCountryCodeChange}
+            className="oxygen-select"
+            inputProps={{
+              className: 'oxygen-select-input-root',
+            }}
+            {...SelectProps}
+          >
+            {countries?.map((country: Country) => (
+              <MenuItem value={country.dial_code} className="oxygen-dial-code-menu-item">
+                <ListItemIcon>
+                  <Image
+                    loading="lazy"
+                    src={`https://flagcdn.com/${country.code.toLowerCase()}.svg`}
+                    srcSet={`https://flagcdn.com/${country.code.toLowerCase()}.svg 2x`}
```

Comment:
Using a CDN in a library does not seem right to me. Can we try using flag emojis?
https://dev.to/jorik/country-code-to-flag-emoji-a21

Reply:
Emojis might have some issues in Windows. Maybe we can try a SVG library.

Reply:
Added [react-world-flags](https://www.npmjs.com/package/react-world-flags?activeTab=readme) package